src/App.tsx

```diff
@@ -1,223 +1,244 @@
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext, useEffect, useRef, useState,
+} from 'react';
+import { CSSTransition, TransitionGroup } from 'react-transition-group';
+import * as api from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { TodoItem } from './components/Auth/TodoItem/TodoItem';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [isCreating, setIsCreating] = useState(false);
+  const [processing, setProcessing] = useState<number[]>([]);
+  const [filterBy, setFilterBy] = useState('All');
+  const [error, setError] = useState('');
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
+    if (user) {
+      api.getTodos(user.id).then(response => {
+        setTodos(response);
+      });
+    }
+
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
   }, []);
 
+  const addProcessing = (id: number) => {
+    setProcessing(prevstate => [...prevstate, id]);
+  };
+
+  const removeProcessing = (id: number) => {
+    setProcessing(prevstate => prevstate.filter(item => item !== id));
+  };
+
+  if (!user) {
+    return <p>Please Login</p>;
+  }
+
+  const visibleTodos = todos.filter(todo => {
+    switch (filterBy) {
+      case 'Active':
+        return !todo.completed;
+      case 'Completed':
+        return todo.completed;
+      case 'All':
+      default:
+        return todo;
+    }
+  });
+
+  const activeTodos = todos.filter(todo => !todo.completed);
+  const completedTodos = todos.filter(todo => todo.completed);
+
+  const onDelete = (id: number) => {
+    addProcessing(id);
+    api.deleteTodo(id).then(() => {
+      setTodos(prevstate => prevstate.filter(todo => todo.id !== id));
+    }).catch(() => setError('Failed to delete'))
+      .finally(() => removeProcessing(id));
+  };
+
+  const patchTodo = (newTodo: Todo) => {
+    addProcessing(newTodo.id);
+    api.patchTodo(newTodo).then(() => {
+      setTodos(prevstate => prevstate
+        .map(todo => (todo.id === newTodo.id ? newTodo : todo)));
+    }).catch(() => setError('Failed to update'))
+      .finally(() => removeProcessing(newTodo.id));
+  };
+
+  const toggleAll = () => {
+    let toToggle;
+
+    if (activeTodos.length === 0) {
+      toToggle = completedTodos;
+    } else {
+      toToggle = activeTodos;
+    }
+
+    toToggle.forEach(todo => {
+      patchTodo({ ...todo, completed: !todo.completed });
+    });
+  };
+
+  const clearCompleted = () => {
+    completedTodos.forEach(todo => {
+      onDelete(todo.id);
+    });
+  };
+
+  const onSubmit = () => {
+    if (todoTitle.length === 0) {
+      return;
+    }
```

Comment:
```suggestion
    if (todoTitle.trim.length === 0) {
      return;
    }
```
<img width="523" alt="image" src="https://user-images.githubusercontent.com/55802630/187160069-b49ebe46-bdf3-494c-b91b-e1397f6052bd.png">